src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -431,15 +437,23 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
       this.superinterfaces += superinterfaces
     }
 
-    fun addSuperinterface(superinterface: TypeName) = apply {
-      superinterfaces += superinterface
+    fun addSuperinterface(superinterface: TypeName, delegate: CodeBlock = CodeBlock.empty) = apply {
+      if (delegate === CodeBlock.empty)
```

Comment:
This check will fail if somebody passes a different `CodeBlock.of("")`. Would it be better to use `==`, or check via `delegate.isEmpty()`?

Reply:
Checked with `===` and made the `CodeBlock.EMPTY` internal because I thought it'd be consistent with the way the current API handles `CodeBlock` with zero checks except the for the pre-defined symbols  (i.e. kotlinpoet doesn't stop you from shooting yourself in the foot). Although an empty codeblock is a pretty simple case

Reply:
I changed it to delegate.isEmpty(), so it simply adds the regular superinterface on an empty CodeBlock (doesn't fail it), assuming that's what you meant